### PR TITLE
Jv remove rotten endpoints

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
## Description

Orphaned listening endpoints in sensor that have no associated container are not currently being removed. This fixes that problem.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

CI is sufficient.

### CODE REVIEW

File: sensor/common/networkflow/manager/manager_impl.go [hunk 1]
```diff
@@ -530,7 +530,7 @@ func (m *networkFlowManager) enrichHostContainerEndpoints(hostConns *hostConnect
 		m.enrichContainerEndpoint(&ep, status, enrichedEndpoints)
 		// If processes listening on ports is enabled, it has to be used there as well before being deleted.
 		used := status.used && (status.usedProcess || !env.ProcessesListeningOnPort.BooleanSetting())
-		if used && status.lastSeen != timestamp.InfiniteFuture {
```
Comment (robbycochran) [inline sensor/common/networkflow/manager/manager_impl.go:533]: It looks like we have been marking endpoints with `rotten=true` but not deleting them. I'm assuming that was an oversight in this [PR](https://github.com/stackrox/rox/pull/8425).

**Comment by robbycochran** [comment]
@JoukoVirtanen Have you confirmed that this fixes the HostConnections memory issue in scale tests?

**Review by robbycochran — APPROVED** [review]

File: sensor/common/networkflow/manager/manager_impl.go (hunk 1 above)
Comment (JoukoVirtanen) [reply sensor/common/networkflow/manager/manager_impl.go:533]: Yes. It does look like an oversight from that PR.

**Comment by JoukoVirtanen** [comment]
Two long running clusters were started with fake workload. One using commit 
3.74.x-716-g194152deb2 from the master branch. The other using commit 3.74.x-718-g92014ee659 from the branch jv-remove-rotten-endpoints. The memory usage of sensor was monitored. The memory usage of sensor in master had an upward trend, whereas the memory usage of jv-remove-rotten-endpoints was roughly constant.

Screenshot from 2023-04-14 08-17-59

**Comment by robbycochran** [comment]
> 

Looks good.

**Pull request merged**